app/main.py

```diff
@@ -1,6 +1,19 @@
-from typing import Callable
+from typing import Callable, Any
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_storage = {}
+
+    @wraps(func)
+    def wrapper(*args) -> Any:
+        if args in cache_storage:
```

Comment:
Using mutable types like lists as keys in a dictionary will raise a TypeError because they are not hashable. Consider converting the arguments to a hashable type, such as a tuple, before using them as keys in the cache_storage.